base/compiler/optimize.jl

```diff
@@ -509,7 +513,7 @@ function finish(interp::AbstractInterpreter, opt::OptimizationState,
                     cost_threshold += 4*default
                 end
             end
-            src.inlineable = inline_worthy(ir, params, union_penalties, cost_threshold)
+            src.inlining_cost = UInt16(inline_worthy(ir, params, union_penalties, cost_threshold))
```

Comment:
```suggestion
            src.inlining_cost = inline_worthy(ir, params, union_penalties, cost_threshold)
```

Reply:
I still find the explicit conversion here a "code-smell". Either apply Jameson's suggested change or add a comment why this won't throw

Reply:
Ah, my mistake, this conversion was meant to be removed